application_sdk/outputs/json.py

```diff
@@ -183,22 +187,26 @@ async def write_daft_dataframe(
         # Daft does not have a built in method to write the daft dataframe to json
         # So we are using orjson to write the data to json in a more memory efficient way
         buffer = []
-
-        if isinstance(dataframe, AsyncIterator):
-            async for row in dataframe:
-                self.total_record_count += 1
-                # Serialize the row and add it to the buffer
-                buffer.append(
-                    orjson.dumps(row, option=orjson.OPT_APPEND_NEWLINE).decode("utf-8")
+        for row in dataframe.iter_rows():
+            self.total_record_count += 1
+            # Remove null attributes from the row recursively, preserving specified fields
+            cleaned_row = self.process_null_fields(
+                row, preserve_fields, null_to_empty_dict_fields
+            )
+            # Serialize the row and add it to the buffer
+            buffer.append(
+                orjson.dumps(cleaned_row, option=orjson.OPT_APPEND_NEWLINE).decode(
+                    "utf-8"
                 )
+            )
```

Comment:
did we compare how this impacts performance? can we get quick numbers for 100K and 1M record runs?